projects/observability/src/shared/components/cartesian/d3/legend/cartesian-legend.ts

```diff
@@ -161,8 +162,14 @@ export class CartesianLegend<TData> {
       .text(series => series.name)
       .on('click', series => (this.series.length > 1 ? this.updateActiveSeries(series) : undefined));
 
+    this.drawFilterControl(legendEntry.node()!).location.nativeElement.classList.add('filter');
+
     this.updateLegendClassesAndStyle();
 
+    legendEntry
+      .on('mouseover', () => legendEntry.select('.filter').style('visibility', 'visible'))
+      .on('mouseout', () => legendEntry.select('.filter').style('visibility', 'hidden'));
+
```

Comment:
Nitpick: codecov mentions that we don't have any coverage on L170-171. I don';t think it would hurt to create a test for this class directly.